### DISCUSSION TITLE
Disabling compilation for tests, but only on CI

### DISCUSSION THREAD

**Discussion by davidwessman** [shakacode/shakapacker#402] [Q&A]
Hello!

I want to make sure that during CI, we only rely on the precompiled assets and not compiling anything when the tests are running.
However, when tests locally I would like it to recompile the assets to avoid testing with stale assets (since there is no forced precompilation like we do in CI).

I tried adding a separate `test_ci` environment in shakapacker.yaml but that did not work well.
I have not been able to find any environment variable or such I can use, how would you go about this?

My config is very similar to this:
https://github.com/shakacode/shakapacker/blob/master/lib/install/config/shakapacker.yml#L106

**Comment by tomdracz** [comment]
Adding separate environment in `shakapacker.yml` should work as long as that environment is also defined in the Rails app as far as I can see through the code.

One option would be to have `shakapacker_ci.yml` or similar that sets `compile: false` and then on CI pass it using env variable `SHAKAPACKER_CONFIG` to point at this overridden config

**Reply by davidwessman** [reply]
😮 the separate file sounds like a nice solution. Thanks 😊
